src/redis_index.c

```diff
@@ -457,9 +454,16 @@ int Redis_DropIndex(RedisSearchCtx *ctx, int deleteDocuments) {
 
     DocTable *dt = &ctx->spec->docs;
 
-    for (size_t i = 1; i < dt->size; i++) {
-      const RSDocumentMetadata *dmd = dt->docs + i;
-      Redis_DeleteKey(ctx->redisCtx, DMD_CreateKeyString(dmd, ctx->redisCtx));
+    for (size_t i = 1; i < dt->cap; ++i) {
+      DMDChain *chain = &dt->buckets[i];
+      if (DMDChain_IsEmpty(chain)) {
```

Comment:
Abstract this to some kind of function in doc_table.c. None of the code other than that file needs to know how the table is structured.